pip/commands/view.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import os
+import subprocess
+
+from pip.basecommand import Command
+from pip.commands.show import search_packages_info
+from pip.status_codes import SUCCESS, ERROR
+from pip._vendor import pkg_resources
+
+logger = logging.getLogger(__name__)
+
+
+class ViewCommand(Command):
+
+    """
+    Views the package source directory with the editor defined in
+    $PIP_EDITOR.
+    """
+    name = 'view'
+    usage = """
+      %prog <package>"""
+    summary = 'View installed package in the editor'
+
+    def __init__(self, *args, **kw):
+        super(ViewCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        if not args:
+            logger.warning('ERROR: Please provide a package name or names.')
+            return ERROR
+        if not os.getenv('PIP_EDITOR'):
```

Comment:
Maybe as well rather than instead, some people may want to use a different editor on pip. Actually, the editor should be handled as a configuration option that can be set as a command line argument, or in ```pip.ini```, or via environment variables. See the various other options for examples of how this would be done.